### PR TITLE
Sync dimensions

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: apps/fyle/views.py [hunk 1]
```diff
@@ -384,3 +385,71 @@ def get_queryset(self):
...
+                status=status.HTTP_200_OK
+            )
+
+        except FyleCredential.DoesNotExist:
+            return Response(
+                data={
+                    'message': 'Fyle credentials not found in workspace'
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
```
Comment (ashwin1111) [inline apps/fyle/views.py:455]: new line

File: apps/sage_intacct/utils.py [hunk 2]
```diff
@@ -257,6 +257,58 @@ def sync_employees(self):
             employee_attributes, self.workspace_id)
         return account_attributes
 
+    def sync_dimensions(self, workspace_id: str):
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:260]: why is workspace id needed?

File: apps/sage_intacct/views.py [hunk 3]
```diff
@@ -657,3 +659,71 @@ def post(self, request, *args, **kwargs):
...
+        try:
+            workspace = Workspace.objects.get(id=kwargs['workspace_id'])
+            if workspace.destination_synced_at:
+                time_interval = datetime.now(timezone.utc) - workspace.destination_synced_at
+
+            if workspace.destination_synced_at is None or time_interval.days > 0:
+                sage_intacct_credentials = SageIntacctCredential.objects.get(workspace_id=kwargs['workspace_id'])
+                sage_intacct_connecter = SageIntacctConnector(sage_intacct_credentials, workspace_id=kwargs['workspace_id'])
+
+                sage_intacct_connecter.sync_dimensions(kwargs['workspace_id'])
```
Comment (ashwin1111) [inline apps/sage_intacct/views.py:680]: why is workspace id needed?

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: apps/sage_intacct/utils.py (hunk 2 above)
Comment (NileshPant1999) [reply apps/sage_intacct/utils.py:260]: we need to pass it while syncing vendors and accounts

File: apps/sage_intacct/views.py (hunk 3 above)
Comment (NileshPant1999) [reply apps/sage_intacct/views.py:680]: needed while syncing vendor and accounts

File: apps/sage_intacct/utils.py [hunk 4]
```diff
@@ -64,14 +61,14 @@ def sync_accounts(self, workspace_id):
...
 
-            if general_settings and general_settings.corporate_credit_card_expenses_object:
-                account_attributes.append({
-                    'attribute_type': 'CCC_ACCOUNT',
-                    'active': True if account['STATUS'] == 'active' else None,
-                    'display_name': 'Credit Card Account',
-                    'value': unidecode.unidecode(u'{0}'.format(account['TITLE'].replace('/', '-'))),
-                    'destination_id': account['ACCOUNTNO']
-                })
+
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:64]: remove workspace_id in function args, it's not needed now

File: apps/sage_intacct/utils.py [hunk 5]
```diff
@@ -433,14 +427,13 @@ def sync_vendors(self, workspace_id: str):
                 'detail': detail
             })
 
-            if general_settings and general_settings.corporate_credit_card_expenses_object == 'BILL':
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:384]: remove workspace_id in function args, it's not needed now

File: apps/sage_intacct/utils.py (hunk 2 above)
Comment (ashwin1111) [reply apps/sage_intacct/utils.py:257]: not needed now

File: apps/sage_intacct/views.py (hunk 3 above)
Comment (ashwin1111) [reply apps/sage_intacct/views.py:680]: not needed

File: apps/sage_intacct/views.py [hunk 6]
```diff
@@ -657,3 +659,71 @@ def post(self, request, *args, **kwargs):
...
+
+    def post(self, request, *args, **kwargs):
+        """
+        Sync data from sage intacct
+        """
+        try:
+            sage_intacct_credentials = SageIntacctCredential.objects.get(workspace_id=kwargs['workspace_id'])
+            sage_intacct_connecter = SageIntacctConnector(sage_intacct_credentials, workspace_id=kwargs['workspace_id'])
+
+            sage_intacct_connecter.sync_dimensions(kwargs['workspace_id'])
```
Comment (ashwin1111) [inline apps/sage_intacct/views.py:711]: workspace_id not neeed

File: apps/sage_intacct/views.py [hunk 7]
```diff
@@ -657,3 +658,71 @@ def post(self, request, *args, **kwargs):
...
+    def post(self, request, *args, **kwargs):
+
+        try:
+            workspace = Workspace.objects.get(id=kwargs['workspace_id'])
+            if workspace.destination_synced_at:
+                time_interval = datetime.now(timezone.utc) - workspace.destination_synced_at
+
+            if workspace.destination_synced_at is None or time_interval.days > 0:
+                sage_intacct_credentials = SageIntacctCredential.objects.get(workspace_id=kwargs['workspace_id'])
+                sage_intacct_connecter = SageIntacctConnector(sage_intacct_credentials, workspace_id=kwargs['workspace_id'])
```
Comment (ashwin1111) [inline apps/sage_intacct/views.py:677]: sage_intacct_connector

File: apps/sage_intacct/views.py [hunk 8]
```diff
@@ -657,3 +658,71 @@ def post(self, request, *args, **kwargs):
...
+    Refresh Sage Intacct Dimensions view
+    """
+
+    def post(self, request, *args, **kwargs):
+        """
+        Sync data from sage intacct
+        """
+        try:
+            sage_intacct_credentials = SageIntacctCredential.objects.get(workspace_id=kwargs['workspace_id'])
+            sage_intacct_connecter = SageIntacctConnector(sage_intacct_credentials, workspace_id=kwargs['workspace_id'])
```
Comment (ashwin1111) [inline apps/sage_intacct/views.py:708]: sage_intacct_connector

**Review by ashwin1111 — APPROVED** [review]

File: apps/mappings/tasks.py [hunk 9]
```diff
@@ -386,7 +386,7 @@ def auto_create_category_mappings(workspace_id):
     corporate_credit_card_expenses_object = general_settings.corporate_credit_card_expenses_object
 
     category_mappings = []
-
+
```
Comment (Sravanksk) [inline apps/mappings/tasks.py:389]: remove this extra tab space

**Review by Sravanksk — APPROVED** [review]

**Pull request merged**